### PR TITLE
Vscode extensions updates & additions

### PR DESCRIPTION
**Pull request by dcarosone**
###### Motivation for this change

Update some vscode extensions

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/vscode-extensions/default.nix [hunk 1]
```diff
@@ -384,6 +384,20 @@ let
...
 
+      cweijan.vscode-database-client2 = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "vscode-database-client2";
+          publisher = "cweijan";
+          version = "4.3.3";
+          sha256 = "06bj23y5rbpz0lw45p1sxssalgn19iqfqbijw2ym22grm0i9hcby";
+        };
+        meta = {
+          description = " Database Client For Visual Studio Code";
```
Comment (bobby285271) [inline pkgs/misc/vscode-extensions/default.nix:395]:
```suggestion
          description = "Database Client For Visual Studio Code";
```

Comment (dcarosone) [reply pkgs/misc/vscode-extensions/default.nix]: Thanks, addressed

**Comment by bobby285271** [comment]
Please fixup the commits and follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when writing commit message (for package additions, use `pkg-name: init at xxx`), and many committers here merge PRs via merge commit and actually we don't require PR to be up to date (if there is no merge conflict).

**Comment by bobby285271** [comment]
I took the liberty to fixup the commits. Many committers here actually merge PRs via merge commit and we don't require the PRs to be up-to-date if CI is happy and there is no merge conflict. And it will always be nice to follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) for commit messages.

**Comment by bobby285271** [comment]
@ofborg build vscode-extensions.mechatroner.rainbow-csv vscode-extensions.serayuzgur.crates vscode-extensions.usernamehw.errorlens vscode-extensions.tuttieee.emacs-mcx vscode-extensions.cweijan.vscode-database-client2

**Pull request merged**